core/src/subgraph/context/instance.rs

```diff
@@ -155,7 +156,36 @@ where
         })
     }
 
-    pub(super) fn revert_data_sources(&mut self, reverted_block: BlockNumber) {
+    /// Reverts any DataSources that have been added from the block forwards (inclusively)
+    /// This function also reverts the done_at status if it was 'done' on this block or later.
+    /// It only returns the offchain::Source because we don't currently need to know which
+    /// DataSources were removed, the source is used so that the offchain DDS can be found again.
+    pub(super) fn revert_data_sources(
+        &mut self,
+        reverted_block: BlockNumber,
+    ) -> Vec<offchain::Source> {
+        self.revert_quick(reverted_block);
+
+        // The following code handles resetting offchain datasources so in most
+        // cases this is enough processing.
+        // At some point we prolly need to improve the linear search but for now this
+        // should be fine. *IT'S FINE*
+        if self.causality_region_seq.0 == CausalityRegion::ONCHAIN {
```

Comment:
The comment could better explain why this check guarantees that there are no offchain data sources. Perhaps it would help readability to add a method to the sequence such as `.is_minimum()`. Then we can keep the inner value in `CausalityRegionSeq` private.

Reply:
I agree with the comment, disagree with making it private. having minimum or whatever is opaque and makes the code impossible to understand without a lot of context. Will try to improve the comment but I'd rather leave ONCHAIN exposed so we have this clarity. Otherwise we could do is_onchain or has_offchain type thing.